### PR TITLE
DMP-1552 - Add Annotations to Hearing details screen

### PR DESCRIPTION
**Pull request by joshw-hmcts**
### Jira link (if applicable)

[DMP-1552](https://tools.hmcts.net/jira/browse/DMP-1552)

### Change description ###
- Adds annotations tab to Hearing screen
- Adds service to hearing.service to get annotations for that hearing

### Checklist

- [X] commit messages are meaningful and follow good commit message guidelines
- [X] README and other documentation has been updated / added (if needed)
- [X] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: darts-api-stub/annotations/annotation.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  type: 'CASE_104',
+  title: 'The requested document cannot be found',
+  status: 404,
+};
+
+const getAnnotationDocumentByAnnotationAndDocumentId = (annotationId, annotationDocumentId) => {
+  const annotations = localArray('annotations');
+  const annotation = annotations.value.find((annotation) => annotation.annotation_id === parseInt(annotationId));
+  if (!annotation) return null;
+  return annotation.annotation_documents.find(
```
Comment (bpires2) [inline darts-api-stub/annotations/annotation.js:17]: shouldn't this just be `return annotation` local variable?

File: darts-api-stub/hearings/hearings.js [hunk 2]
```diff
@@ -1,7 +1,14 @@
 const express = require('express');
+const { localArray } = require('../localArray');
```
Comment (bpires2) [inline darts-api-stub/hearings/hearings.js:2]: localArray sounds confusing, could we call this something like local JSON data?

File: src/app/portal/components/hearing/hearing.component.ts [hunk 3]
```diff
@@ -109,6 +122,7 @@ export class HearingComponent implements OnInit {
   case$ = this.caseService.getCase(this.caseId).pipe(shareReplay(1));
   hearing$ = this.caseService.getHearingById(this.caseId, this.hearingId);
   audio$ = this.hearingService.getAudio(this.hearingId);
+  annotations$ = this.hearingService.getAnnotations(this.hearingId);
```
Comment (bpires2) [inline src/app/portal/components/hearing/hearing.component.ts:125]: I think you need to prevent the observable call here if the user is not a judge or admin to prevent a 403 returning from the backend to error out the application

File: src/app/portal/services/annotation/annotation.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import { of } from 'rxjs';
+import { AnnotationService } from '@services/annotation/annotation.service';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { TestBed } from '@angular/core/testing';
+
+describe('HearingService', () => {
```
Comment (bpires2) [inline src/app/portal/services/annotation/annotation.service.spec.ts:6]: I don't think this is quite right 😅

File: src/app/portal/services/mapping/mapping.service.spec.ts [hunk 5]
```diff
@@ -13,4 +16,74 @@ describe('MappingService', () => {
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
+
+  describe('#mapTranscriptRequestToRows', () => {
+    it('should map correctly', () => {
+      const hearingDateDateTime = DateTime.fromISO('2024-02-01');
+      const requestedOnDateTime = DateTime.fromISO('2024-01-01');
```
Comment (bpires2) [inline src/app/portal/services/mapping/mapping.service.spec.ts:23]: I think it may be better if these variables were removed and put in-line with the properties to make it more readable

**Review by bpires2 — CHANGES_REQUESTED** [review]

File: darts-api-stub/annotations/annotation.js (hunk 1 above)
Comment (joshw-hmcts) [reply darts-api-stub/annotations/annotation.js:17]: This line is looking for the document ID in the same annotation it found in the first find. If either the annotation or document are not found it will return `undefined` which will satisfy the check on line 28 and return a 404

File: cypress/e2e/functional/case-retention-spec.cy.js [hunk 6]
```diff
@@ -130,12 +130,15 @@ describe('Case retention screen as standard user', () => {
       cy.get('.govuk-error-message').should('exist');
       cy.get('.govuk-error-summary').should(
         'contain',
-        'You do not have permission to reduce the current retention date. Please refer to the DARTS retention policy guidance'
+        'You do not have permission to reduce the current retention date'
```
Comment (davet1985) [inline cypress/e2e/functional/case-retention-spec.cy.js:133]: Are these changes related to the ticket?

File: darts-api-stub/annotations/annotation.js (hunk 1 above)
Comment (bpires2) [reply darts-api-stub/annotations/annotation.js:17]: Ah, ok that makes more sense, thank you

File: src/app/portal/components/hearing/hearing.component.ts (hunk 3 above)
Comment (joshw-hmcts) [reply src/app/portal/components/hearing/hearing.component.ts]: Now fixed

File: cypress/e2e/functional/case-retention-spec.cy.js (hunk 6 above)
Comment (joshw-hmcts) [reply cypress/e2e/functional/case-retention-spec.cy.js]: Removed and stashed for another branch

File: src/app/portal/services/annotation/annotation.service.spec.ts (hunk 4 above)
Comment (joshw-hmcts) [reply src/app/portal/services/annotation/annotation.service.spec.ts]: Fixed

File: src/app/portal/services/mapping/mapping.service.spec.ts (hunk 5 above)
Comment (joshw-hmcts) [reply src/app/portal/services/mapping/mapping.service.spec.ts]: Fixed

File: src/app/portal/services/mapping/mapping.service.spec.ts [hunk 7]
```diff
@@ -13,4 +16,74 @@ describe('MappingService', () => {
...
+          status: 'Complete',
+          type: 'TYPE',
+        },
+      ]);
+    });
+  });
+
+  describe('#mapAnnotationsDataToAnnotations', () => {
+    it('should map correctly', () => {
+      const hearingDateIso = '2024-02-01';
```
[the remark this answers is not in the document]
Comment (joshw-hmcts) [reply src/app/portal/services/mapping/mapping.service.spec.ts]: Fixed

**Review by bpires2 — APPROVED** [review]
